teeracle/src/lib.rs

```diff
@@ -51,31 +52,76 @@ pub mod pallet {
 		/// The overarching event type.
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 		type WeightInfo: WeightInfo;
+		/// Max number of whitelisted oracles allowed
+		#[pallet::constant]
+		type MaxOracles: Get<u32>;
 	}
 
+	/// Exchange rates chain's cryptocurrency/currency
 	#[pallet::storage]
 	#[pallet::getter(fn exchange_rate)]
-	pub(super) type ExchangeRates<T> =
+	pub(super) type ExchangeRates<T: Config> =
 		StorageMap<_, Blake2_128Concat, CurrencyString, ExchangeRate, ValueQuery>;
 
+	/// whitelist of trusted exchange rate oracles
+	#[pallet::storage]
+	#[pallet::getter(fn whitelist)]
+	pub(super) type Whitelist<T: Config> =
+		StorageValue<_, WeakBoundedVec<[u8; 32], T::MaxOracles>, ValueQuery>;
+
```

Comment:
I have never heard about this, and couldn't find material. Can you give me some background about the weakbounded vec? :)

Reply:
Found the info, thanks

Reply:
See: https://github.com/paritytech/substrate/blob/master/frame/support/src/storage/weak_bounded_vec.rs
This is similar to BoundedVec, but less strict. It allows some methods to go beyond the vector boundary,  logging in such cases logs error messages, but have methods like try-append, try_mutate, which ensure not to go beyond some length
I used it to ensure the limit of the whitelist size.